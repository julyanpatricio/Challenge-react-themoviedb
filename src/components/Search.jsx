import React, { useState } from "react";
import { useDispatch } from "react-redux";
import { getMoviesByName } from "../actions";
import { getMovies } from "../actions";
import { useHistory } from "react-router-dom";

function Search() {
  const dispatch = useDispatch();
  const [movieSearch, setMovieSearch] = useState({
    name: '',
  });
  const { push } = useHistory()

  function handleChange(e) {
    setMovieSearch(movieSearch => ({
      ...movieSearch,
      [e.target.name]: e.target.value
    }))
  }

  function handleSubmit(e) {
    e.preventDefault()
    movieSearch.name ? dispatch(getMoviesByName(movieSearch.name)) :
      dispatch(getMovies())
    push('/')
  }

  return (
    <form className="form-container" onSubmit={handleSubmit}>
      <input
        id='search-bar'
        type="text"
        name="name"
        autoComplete="off"
        value={movieSearch.name}
        onChange={handleChange} 
      />
      <button type="submit" className='button-search is-green'>Search movies</button>
    </form>
  )
}

export default Search