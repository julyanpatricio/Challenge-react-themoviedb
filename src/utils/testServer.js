import { render } from '@testing-library/react';
import App from '../App';
import React from 'react';
import '../index.css';
import { BrowserRouter } from 'react-router-dom';
import { Provider } from "react-redux";
import store from "../store/index";
import axios from 'axios';

axios.defaults.baseURL = "https://api.themoviedb.org/3"

export const Render = (child) => render(<React.StrictMode>
  <Provider store={store}>
    <BrowserRouter>
      {child}
    </BrowserRouter>
  </Provider>
</React.StrictMode>);