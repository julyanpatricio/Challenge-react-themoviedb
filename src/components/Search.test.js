import React from "react";
import Search from "./Search";
import { Render } from "../utils/testServer";
import '@testing-library/jest-dom'


test("renders movies search", () => {
  const {container} = Render(
      <Search />
  );
  const button = container.querySelector(".button-search")
  expect(button).toBeVisible()
  expect(button).toHaveClass("is-green")
  expect(button).toMatchSnapshot()
});
