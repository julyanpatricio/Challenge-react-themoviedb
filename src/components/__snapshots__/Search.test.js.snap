// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders movies search 1`] = `
<button
  class="button-search is-green"
  type="submit"
>
  Search movies
</button>
`;
